jellyfin-core/src/main/kotlin/org/jellyfin/sdk/discovery/RecommendedServerDiscovery.kt

```diff
@@ -27,103 +29,95 @@ public class RecommendedServerDiscovery(
 		val responseTime: Long,
 	)
 
-	private suspend fun getSystemInfoResult(address: String): SystemInfoResult? {
+	private val logger = LoggerFactory.getLogger("RecommendedServerDiscovery")
+
+	@Suppress("MagicNumber")
+	private fun assignScore(result: SystemInfoResult, parentResult: SystemInfoResult? = null): RecommendedServerInfo {
+		val version = result.systemInfo?.version?.let(ServerVersion::fromString)
+		val score = when {
+			// Did not reply with a valid system information
+			result.systemInfo == null -> RecommendedServerInfoScore.BAD
+			// Wrong product name - might be a different service on this connection
+			!result.systemInfo.productName.equals(PRODUCT_NAME) -> RecommendedServerInfoScore.BAD
+			// Version could not be parsed
+			version == null -> RecommendedServerInfoScore.BAD
+			// Server might be incompatible because it's below the minimum supported server version
+			version < Jellyfin.minimumVersion -> RecommendedServerInfoScore.OK
+			// API might differ slightly but at least above the minimum version
+			version < Jellyfin.apiVersion -> RecommendedServerInfoScore.GOOD
+			// Server is not too fast but should work
+			result.responseTime > SLOW_TIME_THRESHOLD -> RecommendedServerInfoScore.GOOD
+			// All other checks completed, server should have a great connection
+			else -> RecommendedServerInfoScore.GREAT
+		}
+
+		return RecommendedServerInfo(
+			result.address,
+			result.responseTime,
+			score,
+			result.systemInfo,
+			parentResult?.address
+		)
+	}
+
+	private suspend fun getSystemInfoResult(address: String): SystemInfoResult {
 		logger.info("Requesting public system info for $address")
 
-		val client = jellyfin.createApi(baseUrl = address)
+		val client = jellyfin.createApi(
+			baseUrl = address,
+			httpClientOptions = HttpClientOptions(
+				followRedirects = false,
+				timeout = HTTP_TIMEOUT
+			)
+		)
 		val api = SystemApi(client)
 
-		val startTime = System.currentTimeMillis()
-
-		@Suppress("TooGenericExceptionCaught")
-		val info = try {
-			api.getPublicSystemInfo()
-		} catch (err: ConnectException) {
-			logger.debug("Could not connect to $address", err)
-			null
-		} catch (err: Exception) {
-			logger.error("Could not retrieve public system info for $address", err)
-			null
+		val info: Response<PublicSystemInfo>?
+		val responseTime = measureTimeMillis {
```

Comment:
Note for later: when `measureTimedValue` is not experimental anymore, that could be used instead.

Reply:
True, although the calculation later only needs the milliseconds anyway

Reply:
I meant that you could include `info` with that as well :P
But yeah, it doesn't matter for now.